File: Dicoding-rizky92/MADE/MADEDicodingSubmission1/app/src/main/java/com/rizky92/madedicodingsubmission1/MainActivity.java
package com.rizky92.madedicodingsubmission1;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    private Adapter adapter;
    private ArrayList<Movies> list = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final ListView listView = findViewById(R.id.lv_movies);
        adapter = new Adapter(this);
        listView.setAdapter(adapter);

        list.addAll(Data.getListData());

        adapter.setMovies(list);

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                Intent intent = new Intent(view.getContext(), DetailActivity.class);
                Movies movies = (Movies) adapter.getItem(i);
                intent.putExtra(DetailActivity.EXTRA_MOVIES, movies);
                startActivity(intent);
            }
        });
    }
}

File: Dicoding-rizky92/MADE/LatihanListView-kotlined/app/src/main/java/com/rizky92/latihanlistview/MainActivity.kt
package com.rizky92.latihanlistview

import androidx.appcompat.app.AppCompatActivity

import android.content.res.TypedArray
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.Toast

import java.util.ArrayList

class MainActivity : AppCompatActivity() {
    private var dataName: Array<String>? = null
    private var dataDesc: Array<String>? = null
    private var dataFoto: TypedArray? = null
    private var heroAdapter: HeroAdapter? = null
    private var heroes: ArrayList<Hero>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val listView = findViewById<ListView>(R.id.list_view)
        heroAdapter = HeroAdapter(this)
        listView.adapter = heroAdapter

        prepare()
        addItem()

        listView.onItemClickListener = AdapterView.OnItemClickListener {
            adapterView, view, i, l -> Toast.makeText(this@MainActivity, heroes!![i].name, Toast.LENGTH_SHORT).show()
        }
    }

    private fun prepare() {
        dataName = resources.getStringArray(R.array.data_name)
        dataDesc = resources.getStringArray(R.array.data_description)
        dataFoto = resources.obtainTypedArray(R.array.data_photo)
    }

    private fun addItem() {
        heroes = ArrayList()

        for (i in dataName!!.indices) {
            val hero = Hero()

            hero.foto = dataFoto!!.getResourceId(i, -1)
            hero.name = dataName!![i]
            hero.desc = dataDesc!![i]

            heroes!!.add(hero)
        }
        heroAdapter!!.setHeroes(heroes)
    }
}
